doc.go

```diff
@@ -63,58 +64,61 @@ Start skipper and make an HTTP request through skipper:
 Routing Mechanism
 
 The core of skipper's request processing is implemented by a reverse
-proxy in the 'proxy' package. The proxy takes the incoming request,
-passes it to the routing engine to receive the best matching route. If a
-route is found, the request is passed to all filters defined by it. The
-filters can manipulate the request or execute any other logic. Once the
-request is processed by all the filters, it is forwarded to the backend
-endpoint of the route. The response from the backend is passed again to
-all the filters, but in reverse order, and then finally mapped as the
-response to the original incoming request.
+proxy in the 'proxy' package. The proxy receives the incoming request,
+forwards it to the routing engine in order to receive the most closely
+matching route. When a route has been found, the request is forwarded
+to all filters defined by it. The filters can modify the request or
+execute any kind of program logic. Once the request has been processed
+by all the filters, it is forwarded to the backend endpoint of the
+route. The response from the backend goes once again through all the
+filters, but this time the order is reversed, then it finally gets
+mapped as the response of the original incoming request.
 
 Besides the default proxying mechanism, it is possible to define routes
 without a real network backend endpoint. This is called a 'shunt'
 backend, in which case one of the filters needs to handle the request
-(e.g. the 'static' filter).
+(e.g. the 'static' filter). Actually, filters themselves can instruct
+the request flow to shunt.
 
-For details, see the documentation of the proxy subdirectory.
+For further details, see the 'proxy' and 'filters' package
+documentation.
 
 
 Matching Requests
 
-Finding the route for a request happens by matching the request
-attributes against the conditions in the route definitions. Route
-definitions may have the following conditions: method, path (optionally
-with wildcards), path regular expressions, host regular expressions,
-headers and header regular expressions. Besides, it is possible to
-define custom predicates, implemented as extensions.
+Finding a request's route happens by matching the request attributes to
+the conditions in the route's definitions. Such definitions may have the
+following conditions: method, path (optionally with wildcards), path
+regular expressions, host regular expressions, headers and header
+regular expressions. There is also a way to use custom predicates as
+conditions to be matched.
 
```

Comment:
I'd suggest breaking this into a, more readable, list:
"- HTTP Method
- Path (supports wildcards and regular expressions)
- Host header (also supports regular expressions)
- Other HTTP headers (also supports regular expressions)"

"It is also possible to create custom predicates with any other matching criteria"